tests/method-chain/__snapshots__/jsfmt.spec.js.snap

```diff
@@ -708,9 +695,287 @@ this.layoutPartsToHide = this.utils.hashset(
+    dataType: 'json',
+  }).then(
+    (data) => {
+      this.setState({ isLoading: false });
+      this.initWidget(data);
+    },
+    (data) => {
+      this.logImpression('foo_fetch_error', data);
+      Flash.error(I18n.t('offline_identity.foo_issue'));
+    },
+  );
+}
+  
+action$.ofType(ActionTypes.SEARCHED_USERS)
+  .map(action => action.payload.query)
+  .filter(q => !!q)
+  .switchMap(q =>
+    Observable.timer(800) // debounce
+      .takeUntil(action$.ofType(ActionTypes.CLEARED_SEARCH_RESULTS))
+      .mergeMap(() =>
+        Observable.merge(
+          Observable.of(replace(\`?q=\${q}\`)),
+          ajax
+            .getJSON(\`https://api.github.com/search/users?q=\${q}\`)
+            .map(res => res.items)
+            .map(receiveUsers)
+        )
+      )
+  );
+
+window.FooClient
+  .setVars({
+    locale: getFooLocale({ page }),
+    authorizationToken: data.token,
+  })
+  .initVerify('foo_container');
+
+it('gets triggered by mouseenter', () => {
+  const wrapper = shallow(<CalendarDay />);
+  wrapper.dive().find(Button).prop();
+});
+
+x.a(true).b(null).c(123)
+x.d('').e(\`\`).f(g)
+x.h(i.j).k(l()).m([n, o]);
+
+=====================================output=====================================
+// examples from https://github.com/prettier/prettier/issues/4125
+
+const sha256 = data => crypto.createHash("sha256").update(data).digest("hex");
+
+req.checkBody("id").isInt().optional();
+req.checkBody("name").notEmpty().optional();
+
+const x = moment().add(1, "day").valueOf();
+
+// should stay on one line:
+obj.foo(1).foo(2).foo(3);
+
+// better on multiple lines:
+somePromise
+  .then(val => doSomething(val))
+  .catch(err => handleError(err));
+
+// you can still force multi-line chaining with a comment:
+const sha256 = data =>
+  crypto // breakme
+    .createHash("sha256")
+    .update(data)
+    .digest("hex");
+
+// examples from https://github.com/prettier/prettier/pull/4765
+
+if ($(el).attr("href").includes("/wiki/")) {
+}
+
+if ($(el).attr("href").includes("/wiki/")) {
+  if ($(el).attr("xyz").includes("/whatever/")) {
+    if ($(el).attr("hello").includes("/world/")) {
+    }
+  }
+}
+
+const parseNumbers = s => s.split("").map(Number).sort();
+
+function palindrome(a, b) {
+  return a.slice().reverse().join(",") === b.slice().sort().join(",");
+}
+
+// examples from https://github.com/prettier/prettier/issues/1565
+
+d3.select("body")
+  .selectAll("p")
+  .data([1, 2, 3])
+  .enter()
+  .style("color", "white");
+
+Object.keys(props)
+  .filter(key => key in own === false)
+  .reduce((a, key) => {
+    a[key] = props[key];
+    return a;
+  }, {});
+
+point().x(4).y(3).z(6).plot();
+
+assert.equal(this.$().text().trim(), "1000");
+
+something()
+  .then(() => doSomethingElse())
+  .then(result => dontForgetThisAsWell(result));
+
+db.branch(
+  db
+    .table("users")
+    .filter({ email })
+    .count(),
+  db.table("users").insert({ email }),
+  db.table("users").filter({ email })
+);
```

Comment:
This inconsistency is somewhat unexpected to me.

Reply:
Overall, the decision to exclude the last chain member from the `isSimple` check looks questionable to me. Also I think simple shortcut object literals like `{ email }` should pass this check.

Reply:
Good point about object literals with shorthand properties, will add them. I'll also see what difference including the last chain member makes, I'm not too tied to it being all but one. I made it that way to start with because the chains can still be pretty readable, but I don't see it making a huge difference.